### ISSUE TITLE
SQL Server create schema with composite key not working

### ISSUE THREAD

**Issue by frevib** [JetBrains/Exposed#1166]
When creating a schema using a composite key, an incorrect SQL query is generated.

Given:
```kotlin
object Temp : Table("dbo.Temp") {
    val someInt = long("someInt")
    val someLong = integer("someLong")
    override val primaryKey = PrimaryKey(someInt, someLong)
}
```

When:
```kotlin
transaction(database) {
    SchemaUtils.create(Temp)
}
```

Then:

```
14:08:32.377 [Test worker] WARN  Exposed - Transaction attempt #0 failed: com.microsoft.sqlserver.jdbc.SQLServerException: Incorrect syntax near '.'.. Statement(s): CREATE TABLE dbo.Temp (someInt BIGINT, someLong INT, CONSTRAINT pk_dbo.Temp PRIMARY KEY (someInt, someLong))
org.jetbrains.exposed.exceptions.ExposedSQLException: com.microsoft.sqlserver.jdbc.SQLServerException: Incorrect syntax near '.'.
```

If `pk_dbo.Temp` is wrapped in double quotes or square brackets, the query syntax is valid. E.g the following is correct syntax: 

```sql
CREATE TABLE dbo.Temp (someInt BIGINT, someLong INT, CONSTRAINT "pk_dbo.Temp" PRIMARY KEY (someInt, someLong))
```

However, the PK should not be prepended to the schema.

**Comment by frevib** [comment]
Bypassed it by stating the PK name explicitly:

```kotlin
override val primaryKey = PrimaryKey(someInt, someLong, name = "pk_Temp")
```

**Comment by dtbullock** [comment]
So the general case would be: if Exposed ever has occasion to derive an identifier, it needs to minimally ensure it is properly delimited, as per the rules for identifiers in that database.  

For SQL Server, that'd be (Classes of Identifiers)[https://docs.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers?view=sql-server-ver15#classes-of-identifiers]

Though I have to say that I'm not really a fan of the ORM deciding the table-structure.

**Comment by bog-walk** [comment]
This issue was fixed by [PR #1536](https://github.com/JetBrains/Exposed/pull/1536) in version [0.39.1](https://github.com/JetBrains/Exposed/blob/main/docs/ChangeLog.md#0391).

In current version 0.42.1, creating the table in the example above (without providing a custom PK name) generates SQL:
```
CREATE TABLE dbo.Temp (someInt BIGINT, someLong INT, CONSTRAINT pk_Temp PRIMARY KEY (someInt, someLong))
```
